### DISCUSSION TITLE
Amount is mixed in with a Credit/Debit Flag

### DISCUSSION THREAD

**Discussion by R-Nabil** [firefly-iii/firefly-iii#7072] [Firefly III]
Hey,

Very new to FirflyIII so i'm still discovering most of it.
FIrst thing i'm struggling with is obviously the data importing. 

My credit cards account have a strange CSV in which one columns mixes three information : Currency, Amount and Credit/Debit Flag. For instance :
31/01/2022,FancyGroceryStore,EUR 1000 DR
31/01/2022,FancyGroceryStoreRefund,EUR 900 CR

I use FIDI to import data, but if I map the last column to Amount, it always goes as Credit. If I map it to Amount(Debit) or Amount(Credit), it's the same issue.

Which mapping should I use, or what type of rule could help me recognising these transactions correctly ?

Thanks in advance for any help

**Comment by JC5** [comment]
Is the amount not positive or negative? Is there nothing to distinguish the two?

**Reply by R-Nabil** [reply]
No indeed. Only showing this CR/DR for credit (positive) Debit (negative). 

Any way I can create my own parser ?

**Reply by JC5** [reply]
I don't understand. Do you have an actual example line?

**Reply by R-Nabil** [reply]
Sorry you mean more examples ? I have put these two in the original thread

31/01/2022,FancyGroceryStore,EUR 1000 DR
31/01/2022,FancyGroceryStoreRefund,EUR 900

**Reply by JC5** [reply]
So what's this CR/DR you're talking about?

**Reply by R-Nabil** [reply]
You see the amount ? EUR 1000 DR

**Reply by JC5** [reply]
Ahhh I thought it was the currency or something lol. 

Sorry, you'll have to teach your bank how to make proper CSV files, and script something together yourself.

**Reply by R-Nabil** [reply]
Teaching my bank, unlikely :) I can try but I’m pretty sure they will not listen to any feedback. 

So there’s no way I can do my own parser with Fidi ?

**Reply by JC5** [reply]
I'm afraid not.

- You can use rules to recognize key words and change the transaction type.
- You can build something using webhooks (coming in 6.0.0)
- You can use the API directly (https://api-docs.firefly-iii.org/)

**Reply by R-Nabil** [reply]
Not clear to me how I could use the rule actually. It’s what I thought first but couldn’t figure it out. Could you point me in the right direction?

**Reply by R-Nabil** [reply]
Any idea on the rule ? It would seem to be the easiest/fastest but not sure how to proceed ?

**Reply by JC5** [reply]
I keep thinking the data importer doesn't recognize the amount properly, does it?

When it does, try a rule with a "description contains" trigger that triggers on the word refund and then use "change transaction type" as action.

**Reply by R-Nabil** [reply]
Actually it does recognise the amount correctly. 

I’ll try to put a rule on description and let you know if that would work
